Relax diff on maintenance_policy.daily_maintenance_window.start_time

michaelbannister opened the pull request:
Fixes #719.
Add a DiffSuppressFunc to start_time.

I can't see if there's a useful way to test this in the unit test and I can't see how I'd trigger this condition through an acceptance test case. Ideas & advice welcome!

rosbo commented on google/utils_test.go:
```diff
@@ -93,3 +93,11 @@ func TestIpCidrRangeDiffSuppress(t *testing.T) {
+func TestRfc3339TimeDiffSuppress(t *testing.T) {
+	old := "2:00"
+	new := "02:00"
+	if rfc3339TimeDiffSuppress("time", old, new, nil) == false {
```
Add also a test case to make sure it doesn't suppress a diff for the following cases:
2:00 and 3:00
02:00 and 3:00
2:00 and 03:00

I suggest you use a table driven test approach like `TestIpCidrRangeDiffSuppress` in this file.

michaelbannister commented:
Added more test cases as requested. I've removed my assumption that the 'new' value will always have a two-digit hour, in case that validation restriction gets lifted (or used somewhere else where it's not enforced).

rosbo commented:
Thanks Michael!
